resources/web/wwi/protoVisualizer/Vrml.js

```diff
@@ -392,14 +392,23 @@ export class SFNode extends SingleValue {
     const nodeX3d = this.value.toX3d(parameterName);
 
     // handle exceptions
+    let name;
+    if (!this.value.isProto)
+      name = this.value.name;
+    else {
+      let baseType = this.value.baseType;
+      while (baseType.isProto)
+        baseType = baseType.baseType;
+      name = baseType.name;
+    }
```

Comment:
this can be avoided by calling `this.value.getBaseNode().name` directly.
However I have to admit I'm a bit perplexed why we need to do so in the first place. x3d encoding [should only occur for basenodes](https://github.com/cyberbotics/webots/blob/11a426186a8a1ea5d7c28353923937c32a791e1a/resources/web/wwi/protoVisualizer/Node.js#L255-L256) anyway, so I fear we shouldn't end up in this situation in the first place. I have the feeling the problem might be elsewhere, I'll investigate a bit